server/src/main/java/com/google/fhir/gateway/AllowedQueriesChecker.java

```diff
@@ -92,6 +92,29 @@ private boolean requestMatches(RequestDetailsReader requestDetails, AllowedQuery
 
     if (!Strings.isNullOrEmpty(entry.getRequestType())
         && !requestDetails.getRequestType().name().equalsIgnoreCase(entry.getRequestType())) {
+
+      if (!requestDetails.getRequestPath().endsWith(ProxyConstants.HTTP_URL_SEPARATOR)
+          && requestDetails.getRequestPath().contains(ProxyConstants.HTTP_URL_SEPARATOR)
+          && entry.getPath().endsWith(AllowedQueriesConfig.MATCHES_ANY_VALUE)
+          && entry
+              .getPath()
+              .equals(
+                  requestDetails
+                          .getRequestPath()
+                          .substring(
+                              0,
+                              requestDetails
+                                  .getRequestPath()
+                                  .lastIndexOf(ProxyConstants.HTTP_URL_SEPARATOR))
+                      + ProxyConstants.HTTP_URL_SEPARATOR
+                      + AllowedQueriesConfig.MATCHES_ANY_VALUE)) {
```

Comment:
What is the motivation for the empty block? If this is just to avoid `return false` in this particular case, please restructure the code and avoid `if/else` (just one `if` block with negated condition should be enough).